src/libs/actions/Report.js

```diff
@@ -379,11 +379,26 @@ function setLocalLastRead(reportID, sequenceNumber) {
 
     // Update the report optimistically
     Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
-        unreadActionCount: 0,
+        unreadActionCount: Math.max(reportMaxSequenceNumbers[reportID] - sequenceNumber, 0),
         lastVisitedTimestamp: Date.now(),
     });
 }
 
+/**
+ * Set the New marker for a given report to the oldest unread message
+ *
+ * @param {Number} reportID
+ */
+function setNewMarkerInitialPosition(reportID) {
```

Comment:
Is the word "initial" in this name significant? I would assume it means that the new marker position can change, but I am not sure how it would change. How does it change outside of this? Can this method be renamed to `setNewMarkerPosition` and used every time the marker position needs set?

Reply:
I like the name, though. I believe it is more descriptive that either alternative.
This method ONLY sets the marker position when the report mounts, so it is the `initial` marker position. It places the marker where it should be based on the number of unread comments (so above the oldest unread comment).

At a later point, a user can manually mark a comment as `unread` effectively moving the marker above it. That does NOT happen through this method.

So I thought the `initial` keyword was important, as that is what this method does. It places the marker in position when you open the report.

Reply:
>the initial keyword was important, as that is what this method does. It places the marker in position when you open the report

"initial" implies the function should be used in some a particular situation (kind of vague to the reader what this would be) and in reality the function has no idea when it will be used it just performs some logic.

>At a later point, a user can manually mark a comment as unread

Seems like it would be clearer to have just one method that did this.

Reply:
I was going to come and say the same things Marc did... so... 👍 !